api/metrics/v1/labels_enforcer.go

```diff
@@ -34,7 +34,7 @@ func WithEnforceTenancyOnQuery(label string) func(http.Handler) http.Handler {
 				return
 			}
 
-			e := injectproxy.NewEnforcer([]*labels.Matcher{{
+			e := injectproxy.NewEnforcer(false, []*labels.Matcher{{
```

Comment:
What is this new parameter set to `false` here? Can you elaborate a bit.

Reply:
This is parameter is for `errorOnReplace` to allow the enforcer to return an error when [`EnforceMatchers`](https://pkg.go.dev/github.com/prometheus-community/prom-label-proxy@v0.4.0/injectproxy#Enforcer.EnforceMatchers) is called. When false, this method would not report a method. However - as far as I could tell - we do not use this method as of yet.

Reply:
Does this preserve backwards compatibility on how this function behaves? 🤔

Reply:
As far as I know. It seems like the default behavior for this is `false`. However, if I'm not mistaken, this was added in v0.4.0 as a new feature.

Reply:
Hmm, we use `EnforceNode` which indirectly calls `EnforceMatchers`.
So `errOnReplace=false` basically allows us to silently replace the matcher in a PromQL expression (either in query or in rules), if it already has the matcher we are enforcing, but with a different value. 

For e.g if we get a query `http_requests_total{tenant_id="foo"}`, and we enforce `tenant_id=bar`, the resultant query getting processed would be `http_requests_total{tenant_id="foo"}`

This was the default before so we haven't changed anything! 🙂